extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -112,15 +120,45 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
 
   /**
    * Triggers template update if opening without active product data.
-   * This happens when the "Shop Now" CTA is clicked.
+   * This happens either when the "Shop Now" CTA is clicked or when a
+   * shopping tag is clicked.
    * @param {boolean} isOpen
    * @private
    */
   onAttachmentStateUpdate_(isOpen) {
     if (!this.isOnActivePage_() || !isOpen) {
       return;
     }
+
     const shoppingData = this.storeService_.get(StateProperty.SHOPPING_DATA);
+
+    const productId =
+      Object.values(shoppingData[this.pageEl_.id])?.length === 1
+        ? Object.values(shoppingData[this.pageEl_.id])[0].productId
+        : shoppingData?.activeProductData?.productId;
+
+    if (productId === '') {
```

Comment:
If the PLP is opened, wouldn't the productId be null (if `activeProductData = null`)?

Reply:
Also if there's only tag in the page, why would `activeProductData` not get set to the correct value? (that way your could remove the complexity in `productId = ...`)